### PR TITLE
Modified Api Controllers to inherite from ControllerBase abstract cla…

### PR DESCRIPTION
**Pull request by MaherJendoubi**
…ss instead of Controller abstract class

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/Controllers/BrandApiController.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public async Task<IActionResult> Get()
         {
             var brandList = await _brandRepository.Query().Where(x => !x.IsDeleted).ToListAsync();
 
-            return Json(brandList);
+            return new JsonResult(brandList);
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/Controllers/BrandApiController.cs]: I think it should be.

  ```cs     
        [HttpGet]
        public async Task<ActionResult<IEnumerable<Brand>>> Get()
        {
            var brandList = await _brandRepository.Query().Where(x => !x.IsDeleted).ToListAsync();
            return brandList;
        }
```

And in the case of return an anonymous class. We have to create a view model for it.

Comment (MaherJendoubi) [reply src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/Controllers/BrandApiController.cs]: Great idea @thiennn !

Comment (MaherJendoubi) [reply src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/Controllers/BrandApiController.cs]: @thiennn It is strange that Api Controllers cannot be decorated with [ApiController]! Have you any idea about it? I have already tested it but it fails.

**Comment by thiennn** [comment]
This PR is out of date. Let close it, however, feel free to make change and reopen it

**Pull request closed without merging**